libkbfs/block_types.go

```diff
@@ -188,10 +188,30 @@ func (fb *FileBlock) NewEmpty() Block {
 
 // DataVersion returns data version for this block.
 func (fb *FileBlock) DataVersion() DataVer {
+	// If this is an indirect block, and none of its children are
```

Comment:
> I'm confused about this -- if there is a top-level indirect block with a bunch of child blocks not marked as direct, and there are no holes, can't that also be FirstValidDataVer?

Only for files written by older versions of the client.  But `DataVersion()` is only used when writing new files, in which case the direct blocks would always be marked as such.